src/ur/ur_driver.cpp

```diff
@@ -378,10 +354,75 @@ bool UrDriver::setToolVoltage(const ToolVoltage voltage)
     return sendScript(cmd.str());
   }
 }
+// Function for e-series robots (Needs both damping factor and gain scaling factor)
+bool UrDriver::startForceMode(const vector6d_t& task_frame, const vector6uint32_t& selection_vector,
+                              const vector6d_t& wrench, const unsigned int type, const vector6d_t& limits,
+                              double damping_factor, double gain_scaling_factor)
+{
+  if (robot_version_.major < 5)
+  {
+    URCL_LOG_ERROR("Force mode gain scaling factor cannot be set on a CB3 robot.");
+    return false;
+  }
+  // Test that the type is either 1, 2 or 3.
+  switch (type)
+  {
+    case 1:
+      break;
+    case 2:
+      break;
+    case 3:
+      break;
+    default:
+      std::stringstream ss;
+      ss << "The type should be 1, 2 or 3. The type is " << type;
+      URCL_LOG_ERROR(ss.str().c_str());
+      return false;
+  }
+  for (unsigned int i = 0; i < selection_vector.size(); ++i)
+  {
+    if (selection_vector[i] > 1)
+    {
+      URCL_LOG_ERROR("The selection vector should only consist of 0's and 1's");
+      return false;
+    }
+  }
+
+  if (damping_factor > 1 || damping_factor < 0)
+  {
+    URCL_LOG_ERROR("The force mode damping factor should be between 0 and 1, both inclusive.");
+    return false;
+  }
+
+  if (gain_scaling_factor < 0 || gain_scaling_factor > 2)
+  {
+    URCL_LOG_ERROR("The force mode gain scaling factor should be between 0 and 2, both inclusive.");
+    return false;
+  }
```

Comment:
Minor nitpicking: Those two if statements have the inverted order of checking the lower and upper boundary. Using the same order would improve readability.